### PR TITLE
Proper update detection and reboot of the admin node

### PR DESCRIPTION
**Pull request by mssola**
- [x] Initial code
- [x] UI: I've taken the same code as in PR #181. @vitoravelino: let's scratch what you did (since it's going to #180), and let's simply add the update & reboot button (dialog) when the `admin` value contains an update (the backend does the same as in #181).
- [x] Update tests
- [x] More tests (coverage < 100%)

### CODE REVIEW

**Comment by mssola** [comment]
Do not review yet 😇

File: app/views/dashboard/_update_admin_modal.html.erb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+        <h4 class="modal-title" id="modal-label">
+          The admin node needs to reboot in order to apply the software update
+        </h4>
+      </div>
+      <div class="modal-body">
+       <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Tempore maiores repellat quia architecto, sunt commodi earum eos, sint dolor quaerat voluptatum! Earum obcaecati distinctio, at explicabo vel similique minus nam.</p>
```
Comment (vitoravelino) [inline app/views/dashboard/_update_admin_modal.html.erb]: @flavio I guess you could help with the detailed message of admin node reboot?

**Comment by mssola** [comment]
> My biggest concerns is targeting all minions with '*' when we might have unassigned minions (but still registered against the salt master)

As explained on IRC, this is something we don't have to fear for this PR, since we are explicitly only tackling assigned minions. That being said, this will surely be a problem in #180, so I'll also add a comment there.

**Comment by ereslibre** [comment]
https://github.com/kubic-project/salt/blob/master/salt/orch/kubernetes.sls#L3 will need to be updated, so we also disable `rebootmgr` on the admin node.

File: app/models/salt_handler/minion_start.rb [hunk 2]
```diff
@@ -20,12 +20,14 @@ def initialize(salt_event)
   def process_event
     parsed_data = salt_event.parsed_data
 
-    # Ignore the ca minion. It shouldn't be used as part of the k8s cluster.
-    return false if parsed_data["id"] == "ca"
+    # Ignore the ca and the dashboard minions. It shouldn't be used as part of
+    # the k8s cluster.
+    id = parsed_data["id"]
+    return false if id == "ca" || id == "admin"
```
Comment (ereslibre) [inline app/models/salt_handler/minion_start.rb]: This could be simplified for future values as `return false if ["ca", "admin"].include? id`

**Comment by mssola** [comment]
> https://github.com/kubic-project/salt/blob/master/salt/orch/kubernetes.sls#L3 will need to be updated, so we also disable rebootmgr on the admin node.

Good catch!

**Pull request merged**